python-package/lightgbm/sklearn.py

```diff
@@ -325,6 +328,13 @@ def __init__(self, boosting_type='gbdt', num_leaves=31, max_depth=-1,
         self._n_classes = None
         self.set_params(**kwargs)
 
+    def __getstate__(self):
+        """Remove un-picklable attributes before serialization."""
+        client = self.__dict__.pop("_client", None)
+        out = copy.deepcopy(self.__dict__)
+        self._client = client
+        return out
```

Comment:
I believe this can be done in Dask estimators, not in parent class.

Reply:
If we do it there, this code would have to be copied 3 times. Because `_DaskLGBMModel` comes second in MRO, it wouldn't be safe to just put this on that mixin. Are you ok with that duplication?

Reply:
Yeah, probably `__getstate__()` is overridden somewhere above in pure scikit-learn branch of inheritance. But we can name this method as `_getstate()` in `_DaskLGBMModel` and call it, so no duplication will be required.
```
def __getstate__(self):
   return self._get_state()
```
Hust like we currently do for all other methods.

Reply:
ok, good suggestion! did that in https://github.com/microsoft/LightGBM/pull/3883/commits/344376bee1a7a3dbe97f7078a782d091d1084c30.

I called it `_lgb_get_state` because I'm nervous about using general-sounding method names in `_DaskLGBMModel` when there are so many classes upstream of `LGBMClassifier`. I'm nervous about some version of scikit-learn introducing a `_get_state` that would override this. We might not catch that in tests because we don't test against many different versions of scikit-learn.

I think I'll propose a PR to do that for other methods on the `_DaskLGBMModel` mixin as well.